src/shared/fileRegistry.ts

```diff
@@ -0,0 +1,205 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger } from './logger/logger'
+import * as pathutils from './utilities/pathUtils'
+import * as path from 'path'
+
+export interface WorkspaceItem<T> {
+    path: string
+    item: T
+}
+
+/**
+ * WorkspaceFileRegistry lets us index files in the current registry. It is used
+ * for CFN templates among other things
+ */
+export abstract class WorkspaceFileRegistry<T> implements vscode.Disposable {
+    private readonly disposables: vscode.Disposable[] = []
+    private _isDisposed: boolean = false
+    private readonly globs: vscode.GlobPattern[] = []
+    private readonly excludedFilePatterns: RegExp[] = []
+    private readonly registryData: Map<string, T> = new Map<string, T>()
+
+    /**
+     * Load in filesystem items, doing any parsing/validaton as required. If it fails, throws
+     * @param path A string with the absolute path to the detected file
+     */
+    protected abstract load(path: string): Promise<T>
+    /**
+     * Registry name for logs
+     */
+    protected abstract registryName: string
```

Comment:
it's a member variable so just `name` is sufficient without any losing of clarity, while avoiding over-specification.
```suggestion
    protected abstract name: string
```

Reply:
Sure, makes sense to drop the registry name anyway since it's no longer just the CFN registry